tests: matrix CI falis

### Problem


`test_copy_all_files` is giving us some errors for other OS platforms:

https://github.com/scikit-package/scikit-package/actions/runs/16611350492/job/46994906571


```
=========================== short test summary info ============================
FAILED tests/test_io.py::test_copy_all_files_bad - AssertionError: Regex pattern did not match.
 Regex: "\\['COMMIT_EDITMSG',\\ 'tutorial\\.rst'\\]\\ already\\ exists\\ in\\ target\\ dir\\ /private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/pytest\\-of\\-runner/pytest\\-0/test_copy_all_files_bad0/package\\-dir/target\\-dir\\-inside\\-package\\-dir\\."
 Input: "['tutorial.rst', 'COMMIT_EDITMSG'] already exists in target dir /private/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/pytest-of-runner/pytest-0/test_copy_all_files_bad0/package-dir/target-dir-inside-package-dir."
========================= 1 failed, 8 passed in 0.48s ==========================
```

### Proposed solution

@ycexiao Could you pls look into the root cause?

--- a/news/fix-test-copy-all-files.rst
+++ b/news/fix-test-copy-all-files.rst
@@ -1,0 +1,23 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* <news item>
+
+**Deprecated:**
+
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* Use simpler assertions in ``test_copy_all_files_bad``.
+
+**Security:**
+
+* <news item>

--- a/tests/test_io.py
+++ b/tests/test_io.py
@@ -1,5 +1,3 @@
-import re
-
 import pytest
 
 from scikit_package.utils.io import copy_all_files
@@ -97,9 +95,9 @@ def test_copy_all_files_bad(user_filesystem):
     duplicate_names = ["COMMIT_EDITMSG", "tutorial.rst"]
     with pytest.raises(
         FileExistsError,
-        match=re.escape(
-            f"{duplicate_names} already exists in target dir "
-            f"{str(target_dir)}."
-        ),
-    ):
+    ) as error:
         copy_all_files(source_dir, target_dir, exists_ok=False)
+        actual_error_message = str(error.value)
+        assert str(target_dir) in actual_error_message
+        for name in duplicate_names:
+            assert name in actual_error_message

